agent/ui_endpoint_test.go

```diff
@@ -1838,6 +1862,42 @@ func TestUIServiceTopology(t *testing.T) {
 				FilteredByACLs: false,
 			},
 		},
+		{
+			name: "cbackend",
+			httpReq: func() *http.Request {
+				req, _ := http.NewRequest("GET", "/v1/internal/ui/service-topology/cbackend?kind=", nil)
+				return req
+			}(),
+			want: &ServiceTopology{
+				Protocol:         "http",
+				TransparentProxy: false,
+				Upstreams:        []*ServiceTopologySummary{},
+				Downstreams: []*ServiceTopologySummary{
+					{
+						ServiceSummary: ServiceSummary{
+							Name:           "cproxy",
+							Datacenter:     "dc1",
+							Tags:           []string{"http", "https"},
+							Nodes:          []string{"cnative", "cnative", "cnative"},
+							InstanceCount:  3,
```

Comment:
Note to reviewers: I have no idea why we have three instances here and I am not sure this is correct (but not a result of my PR anyways). I think I managed to track it down to https://github.com/hashicorp/consul/blob/1bac1544d094ef356787974cd0440783142e53e1/agent/consul/state/catalog.go#L4124 but I am not exactly sure.

Reply:
So it is correct for `combinedServiceNodesTxn` to return what it returns or should the filtering already happen in it? I guess I am a bit at a loss when trying to understand why `cproxy-https` shows up in both cases -- can you shed some light on it?

Reply:
I'm not sure if that is the correct behavior but before your earlier PR #12098 I think all these services would have been filtered out anyways since they were not TProxy. 

It's probably safest to dedupe outside of `combinedServiceNodesTxn` in the `ServiceTopology` method though.

Reply:
Actually, staring at the code a bit more, I think it makes sense to de-dupe _inside_ `combinedServiceNodesTxn`. `checkServiceNodesTxn` with `connect: false` will always return the a service that overlaps with `checkServiceNodesTxn` with `connect: true` due to the indexing.

This bug was hidden because previously we filtered all non-TProxy services.